src/routes/event/list.tsx

```diff
@@ -9,31 +9,93 @@ import {
   Stack,
   TextField,
 } from "@mui/material";
-import { useEffect } from "react";
+import { useEffect, useState } from "react";
 import { useNavigate } from "react-router-dom";
 import { useAuthState } from "../../states/auth/authState";
 import { useEventState } from "../../states/events/eventsState";
+import { useCategoriesState } from "../../states/categories/categoriesState";
+import { useLocationsState } from "../../states/locations/locationsState";
+import { database } from "../../firebase";
+import { get, ref } from "firebase/database";
 
 export default function EventList() {
   const navigate = useNavigate();
+  const [searchName, setSearchName] = useState("");
+  const [selectedLocation, setSelectedLocation] = useState<string | null>(null);
+  const [selectedLocationEvents, setSelectedLocationEvents] = useState<
+    string[]
+  >([]);
+  const [selectedCategory, setSelectedCategory] = useState<string | null>(null);
+  const [selectedCategoryEvents, setSelectedCategoryEvents] = useState<
+    string[]
+  >([]);
 
-  const { allEvents, getAllEvents } = useEventState();
   const { user } = useAuthState();
+  const { allEvents, getAllEvents } = useEventState();
+  const { allLocations, getAllLocations } = useLocationsState();
+  const { allCategories, getAllCategories } = useCategoriesState();
 
   useEffect(() => {
     getAllEvents();
+    getAllLocations();
+    getAllCategories();
   }, []);
+  useEffect(() => {
+    console.log(selectedLocation);
+    if (selectedLocation) {
+      get(ref(database, "location-events/" + selectedLocation)).then(
+        (snapshot) => {
+          if (snapshot.exists()) {
+            const locationEventsResponse = snapshot.val();
+            const newLocationEvents: string[] = [];
+            Object.keys(locationEventsResponse).forEach((key) => {
+              if (locationEventsResponse[key]) {
+                newLocationEvents.push(key);
+              }
+            });
+            console.log(newLocationEvents);
+            setSelectedLocationEvents(newLocationEvents);
+          }
+        }
+      );
+    }
+  }, [selectedLocation]);
+  useEffect(() => {
+    console.log(selectedCategory);
+    if (selectedCategory) {
+      get(ref(database, "category-events/" + selectedCategory)).then(
+        (snapshot) => {
+          if (snapshot.exists()) {
+            const categoryEventsResponse = snapshot.val();
+            const newCategoryEvents: string[] = [];
+            Object.keys(categoryEventsResponse).forEach((key) => {
+              if (categoryEventsResponse[key]) {
+                newCategoryEvents.push(key);
+              }
+            });
+            console.log(newCategoryEvents);
+            setSelectedCategoryEvents(newCategoryEvents);
+          }
+        }
+      );
+    }
+  }, [selectedCategory]);
```

Comment:
If, someone selected a location, and then selects another location while the prev request didn't finish, you have a race condition, 
If the second request finished first, you have a problem. 
But as of now i can't give you a solution for how firebase cancels the same requests, so that only the last created one is the true one.